src/Azure.Functions.Cli/Actions/KubernetesActions/KubernetesDeployAction.cs

```diff
@@ -137,21 +137,25 @@ public override async Task RunAsync()
             }
             else
             {
+                List<Task> tasks = new List<Task>();
                 if (!await KubernetesHelper.NamespaceExists(Namespace))
                 {
                     await KubernetesHelper.CreateNamespace(Namespace);
                 }
 
                 if (shouldBuild)
                 {
-                    await DockerHelpers.DockerPush(resolvedImageName);
+                    // TODO: Join this with the build above so that it can run in parallel to the Kubernetes object create
+                    tasks.Add(DockerHelpers.DockerPush(resolvedImageName, false));
                 }
-
+                // TODO: Convert these to YAML so that we can send them as a single file
```

Comment:
```csharp
var yaml = KubernetesHelper.SerializeResources(resources, OutputSerializationOptions.Yaml);
``` 

should convert them to yaml, then the other overload of `KubectlApply()` should take care of it. I thought applying them one by one would make errors a bit easier to read, but I'm not sure how true that is.